### ISSUE TITLE
CouldBeSwitchStatement gets confused about blocks

### ISSUE THREAD

**Issue by mst-ableton** [CodeNarc/CodeNarc#722]
Hi, thanks for all your work on CodeNarc! The following code yields a `CouldBeSwitchStatement` on CodeNarc version `3.1.0` (this is run on a Jenkins pipeline, hence the `dir()` function, but I'm sure it would work elsewhere too):

```groovy
void doStuff(String platform) {
  if (platform == 'win') {       // CouldBeSwitchStatement is triggered here
    println 'do some Windows stuff'
  } else {
    println 'do some Mac or Linux stuff'
  }

  println 'significant chunk of code common to all platforms'

  dir('subdir') {
    if (platform == 'linux') {
      println 'do some Linux stuff in the subdir'
    } else {
      println 'do some Mac or Windows stuff in the subdir'
    }
  }

  if (platform == 'mac') {
    println 'do some Mac-specific postprocessing'
  }
}
```

I normally trust CodeNarc's warnings about suspicious code style, but I'm not sure reducing everything down to a `switch` statement would help here -- there is quite a bit of shared code used by all platforms in the _middle_ of the function, and most of the platform-specific code is at the beginning or the end.

Suspiciously, removing the `dir('subdir') {}` _or_ the `if (platform == 'mac') {` block makes the error go away. I'm not sure, but this may be related to https://github.com/CodeNarc/CodeNarc/issues/363.

Let me know if I can help debug the issue further or provide more context. Thanks!

**Comment by chrismair** [comment]
Yes, this does look suspiciously like a false positive violation.

> I normally trust CodeNarc's warnings about suspicious code style

Be aware that CodeNarc contains lots of rules, from multiple sources, and motivated from different conventions, styles and sensibilities. Some rules are even contradictory. Choose the subset that make sense for you. I do not use all of the rules in my own projects.

**Comment by chrismair** [comment]
Will be available in CodeNarc 3.2.0
